server.go

```diff
@@ -1583,44 +1581,73 @@ type GnetHTTP struct {
 // The parameter:c has information about the connection such as it's local and remote address.
 // Parameter:out is the return value which is going to be sent back to the client.
 func (es *GnetHTTP) OnOpened(c gnet.Conn) (out []byte, action gnet.Action) {
-	log.Println("OPENED")
 	return
 }
 
 // OnClosed fires when a connection has been closed.
 // The parameter:err is the last known connection error.
 func (es *GnetHTTP) OnClosed(c gnet.Conn, err error) (action gnet.Action) {
-	log.Println("CLOSED")
 	return
 }
 
 // PreWrite fires just before any data is written to any client socket, this event function is usually used to
 // put some code of logging/counting/reporting or any prepositive operations before writing data to client.
 func (es *GnetHTTP) PreWrite() {
-	log.Println("PREWRITE")
 }
 
-// React fires when a connection sends the server data.
-// Call c.Read() or c.ReadN(n) within the parameter:c to read incoming data from client.
-// Parameter:out is the return value which is going to be sent back to the client.
-func (es *GnetHTTP) React(frame []byte, c gnet.Conn) (out []byte, action gnet.Action) {
-	log.Println("REACTING!!!!!")
+type httpServer struct {
+	*gnet.EventServer
+}
+
+type httpCodec struct {
+	fasthttpserver *Server
+}
+
+func (hc *httpCodec) Encode(c gnet.Conn, buf []byte) (out []byte, err error) {
+	return buf, nil
+}
+
+func (hc *httpCodec) Decode(c gnet.Conn) (out []byte, err error) {
+	buf := c.Read()
+	if buf == nil {
+		return
+	}
+
 	gconn := &GnetConn{
 		gnetConn:    c,
-		readBuffer:  bytes.NewBuffer(frame),
+		readBuffer:  bytes.NewBuffer(buf),
 		writeBuffer: &bytes.Buffer{},
-		localAddr:   c.LocalAddr(),
-		remoteAddr:  c.RemoteAddr(),
 	}
-	c.SetContext(gconn)
 
 	//Bypasses the workpoll based implementation...
-	es.fasthttpserver.serveConn(gconn)
+	err = hc.fasthttpserver.serveConn(gconn)
+	if err == nil {
+		c.ResetBuffer()
+		return gconn.writeBuffer.Bytes(), err
+	}
 
-	action = gnet.Close
-	c.ResetBuffer()
+	if err != errHijacked {
+		c.ResetBuffer()
+		return gconn.writeBuffer.Bytes(), err
+	}
+
+	log.Println("request not ready, yet (Partial request?)")
+
+	return nil, err
+}
+
+//OnInitComplete -
+func (es *GnetHTTP) OnInitComplete(srv gnet.Server) (action gnet.Action) {
+	log.Printf("HTTP server is listening on %s (multi-cores: %t, loops: %d)\n",
+		srv.Addr.String(), srv.Multicore, srv.NumEventLoop)
+	return
+}
 
-	return gconn.writeBuffer.Bytes(), action
+//React -
+func (es *GnetHTTP) React(frame []byte, c gnet.Conn) (out []byte, action gnet.Action) {
+	// handle the request
+	out = frame
```

Comment:
This doesn't make sense to me? The documentation says: `Parameter:out is the return value which is going to be sent back to the client.`